plugin/pkg/admission/limitranger/admission.go

```diff
@@ -249,10 +249,15 @@ func limitRequestRatioConstraint(limitType api.LimitType, resourceName api.Resou
 		return fmt.Errorf("%s max limit to request ratio per %s is %s, but no limit is specified or limit is 0.", resourceName, limitType, enforced.String())
 	}
 
-	observedValue := observedLimValue / observedReqValue
+	observedRatio := float64(observedLimValue) / float64(observedReqValue)
+	maxLimitRequestRatio := enforced.Value()
+	if enforced.Value() <= resource.MaxMilliValue {
+		observedRatio = observedRatio * 1000
+		maxLimitRequestRatio = enforced.MilliValue()
+	}
 
-	if observedValue > enforcedValue {
-		return fmt.Errorf("%s max limit to request ratio per %s is %s, but provided ratio is %d.", resourceName, limitType, enforced.String(), observedValue)
+	if observedRatio > float64(maxLimitRequestRatio) {
+		return fmt.Errorf("%s max limit to request ratio per %s is %s, but provided ratio is %f.", resourceName, limitType, enforced.String(), observedRatio)
```

Comment:
The error message will be wrong since it will be scaled by a factor of 1000.

Update the error message to give the float64(observedLimValue) / float64(observedReqValue) result directly.

so error would say 10.1 instead of 10100